BootloaderCorePkg/Tools/PrepareBuildComponentBin.py

```diff
@@ -16,34 +16,38 @@ def Fatal (msg):
     sys.stdout.flush()
     raise Exception (msg)
 
-
-def CloneFspRepo (fsp_dir, fsp_tag):
-    if not os.path.exists(fsp_dir + '/.git'):
-        print ('Cloning Intel FSP repo ...')
-        cmd = 'git clone https://github.com/IntelFsp/FSP.git %s' % fsp_dir
+def CloneRepo (clone_dir, driver_inf):
+    repo, commit = GetRepoAndCommit (driver_inf)
+    if repo == '' or commit == '':
+        Fatal ('Failed to find repo and commit information!')
+
+    if not os.path.exists(clone_dir + '/.git'):
+        print ('Cloning the repo ... %s' % repo)
+        cmd = 'git clone %s %s' % (repo, clone_dir)
         ret = subprocess.call(cmd.split(' '))
         if ret:
-            Fatal ('Failed to clone FSP repo to directory %s !' % fsp_dir)
+            Fatal ('Failed to clone repo to directory %s !' % clone_dir)
         print ('Done\n')
     else:
-        print ('Update Intel FSP repo ...')
+        print ('Update the repo ...')
         cmd = 'git fetch origin master'
-        ret = subprocess.call(cmd.split(' '), cwd=fsp_dir)
+        ret = subprocess.call(cmd.split(' '), cwd=clone_dir)
         if ret:
-            Fatal ('Failed to update FSP repo to directory %s !' % fsp_dir)
+            Fatal ('Failed to update repo in directory %s !' % clone_dir)
         print ('Done\n')
 
-    print ('Checking out Intel FSP commit ...')
+    print ('Checking out specified version ... %s' % commit)
 
-    cmd = 'git checkout %s' % fsp_tag
-    ret = subprocess.call(cmd.split(' '), cwd=fsp_dir)
+    cmd = 'git checkout %s' % commit
+    ret = subprocess.call(cmd.split(' '), cwd=clone_dir)
     if ret:
-        Fatal ('Failed to check out branch !')
+        Fatal ('Failed to check out specified version !')
     print ('Done\n')
 
 
-def CheckFileListExist (copy_list, sbl_dir):
-    exists = True
+def CheckFileListExist (copy_list):
```

Comment:
let us keep interface more generic , do not depends on env variables.

Reply:
OK. will add it back.